### ISSUE TITLE
A bug in loss computation

### ISSUE THREAD

**Issue by Lyken17** [wkentaro/pytorch-fcn#9]
Hi author

First I would like to thank for your sharing. During training, I notice the loss (around 30000) is obviously larger than fcn experiment on caffe(0.4 ~ 3.0). After reading your code, I find the cause. In your code [trainer.py](https://github.com/wkentaro/pytorch-fcn/blob/master/torchfcn/trainer.py#L29), 
```python
    loss = F.nll_loss(log_p, target, weight=weight, size_average=False)
    if size_average:
        loss /= mask.sum().data[0]
```

Note here `mask` is a `torch.ByteTensor`, which means the data is stored in a byte, value from `0~255` (8 bits). So when you sum it up, it will easily lead to overflow (because answer is also stored in `torch.ByteTensor`). 

For example
```python
test1 = Variable(torch.ones(1, 256,256).type('torch.ByteTensor'))
print(test1, test1.sum())

'''
Variable containing:
( 0 ,.,.) = 
   1   1   1  ...    1   1   1
   1   1   1  ...    1   1   1
   1   1   1  ...    1   1   1
     ...       ⋱       ...    
   1   1   1  ...    1   1   1
   1   1   1  ...    1   1   1
   1   1   1  ...    1   1   1
[torch.ByteTensor of size 1x256x256]
 Variable containing:
 0
[torch.ByteTensor of size 1]
'''
```

The correct way should be 
```python
test1 = Variable(torch.ones(1, 256,256).type('torch.ByteTensor'))
print(test1, test1.data.sum())

'''
Variable containing:
( 0 ,.,.) = 
   1   1   1  ...    1   1   1
   1   1   1  ...    1   1   1
   1   1   1  ...    1   1   1
     ...       ⋱       ...    
   1   1   1  ...    1   1   1
   1   1   1  ...    1   1   1
   1   1   1  ...    1   1   1
[torch.ByteTensor of size 1x256x256]
 65536
'''
```

So to compute the loss properly, you need to change `loss /= mask.sum().data[0]` to `loss /= mask.data.sum()`. 

I have created [a pull request](https://github.com/wkentaro/pytorch-fcn/pull/8) to fix this.

**Comment by Lyken17** [comment]
The experiment result after fix
![log](https://cloud.githubusercontent.com/assets/7783214/24077093/109ec344-0bff-11e7-92a1-1ce8e936978b.png)

**Comment by wkentaro** [comment]
Thanks, but the result should not be affected because I use `size_average=False` in trainer.py in actual.
And I think it is same configuration with the original project: https://github.com/shelhamer/fcn.berkeleyvision.org/blob/master/voc-fcn32s/net.py#L64

**Comment by Lyken17** [comment]
Oops, careless me. I forget I have turned on the size_average. But thanks to it, a potential bug is avoided. XD

**Comment by wkentaro** [comment]
Yeah, Thanks!
